Fixed rtl placement for popups

barklund opened the pull request:
## Context

This fixes the popup placement calculation in RTL contexts by flipping some left and right logic in those cases.

As I fixed this, I noticed a bug in the color picker itself (which was never visible before as the color picker itself was never usable in RTL) with the pointer being wrongly aligned - I fixed this too.

## User-facing changes

| Before | After |
|-|-|
| ![Screen Shot 2021-05-04 at 16 56 51](https://user-images.githubusercontent.com/637548/117068955-e4c6cc80-acf9-11eb-965b-e9d6f13e559b.png) | ![Screen Shot 2021-05-04 at 16 54 25](https://user-images.githubusercontent.com/637548/117068959-e5f7f980-acf9-11eb-8c26-8298c94bf560.png) |
| ![Screen Shot 2021-05-04 at 17 08 26](https://user-images.githubusercontent.com/637548/117070245-797dfa00-acfb-11eb-96af-8687e9795ccb.png) | ![Screen Shot 2021-05-04 at 17 07 34](https://user-images.githubusercontent.com/637548/117070256-7aaf2700-acfb-11eb-8c96-98f309069eac.png) |

## Testing Instructions

### QA

This PR can be tested by following these steps:

1. Flip WP admin to RTL
2. Select the background element
3. Open the color picker
4. Observe that it is correctly placed adjacent to but not overlapping the design panel
5. Use the color picker saturation selector to pick a color
6. Observe that the sample circle is correctly placed relative to the selection

### UAT

<!-- ignore-task-list-start -->
- [x] UAT should use the same steps as above.
<!-- ignore-task-list-end -->

## Checklist

<!-- Check these after PR creation -->

- [x] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [x] I have tested this code to the best of my abilities
- [x] I have verified accessibility to the best of my abilities ([docs](https://github.com/google/web-stories-wp/blob/main/docs/accessibility-testing.md))
- [x] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [x] This PR contains automated tests (unit, integration, and/or e2e) to verify the code works as intended ([docs](https://github.com/google/web-stories-wp/tree/main/docs#testing))
- [x] I have added documentation where necessary
- [x] I have added a matching `Type: XYZ` label to the PR

---

Fixes #6804

Commit:
Fixed rtl placement for popups

Commit:
Fixed placement of color picker pointer in RTL

Commit:
Removed irrelevant `defaultProps`

swissspidy commented on assets/src/edit-story/components/colorPicker/pointer.js:
```diff
@@ -60,10 +66,9 @@ const Pointer = styled.div.attrs(({ currentColor, withAlpha }) => {
-};
+function Pointer(props) {
+  const { isRTL } = useConfig();
+  return <PointerElement {...props} isRTL={isRTL} />;
```
```suggestion
  return <PointerElement {...props} $isRTL={isRTL} />;
```

So we don't pollute the DOM with this prop. Of course need to update the styled component above as well.

barklund replied:
`isRTL` is never a valid HTML prop for any element, so it's not relevant. This prop will never pollute the DOM output as React will filter it out as an invalid DOM property.

Transient props are only important for attributes, that *are* valid HTML attributes, but just not used as such. E.g. `title`, `offset`, `background`, `x` etc.

barklund replied:
One could argue that we should use transient naming for all props used only for S-C that aren't supposed to go into the DOM, as a rule, to avoid forgetting it when it's important, but I think that's overkill. I think we should just remember to do this for properties, where this is a problem - and try to remind each other if we forget.

miina commented on assets/src/edit-story/components/colorPicker/pointer.js:
```diff
@@ -42,7 +47,8 @@ const Pointer = styled.div.attrs(({ currentColor, withAlpha }) => {
   transform: translate(
-    ${({ offsetX = 0, offsetY = 0 }) => `${offsetX}px, ${offsetY}px`}
+    ${({ offsetX = 0, offsetY = 0, isRTL }) =>
+      `${(isRTL ? 1 : 1) * offsetX}px, ${offsetY}px`}
```
```suggestion
      `${(isRTL ? -1 : 1) * offsetX}px, ${offsetY}px`}
```

barklund replied:
How did I manage to remove this before pushing? I clearly had this when creating the above screenshots? Weiiiiiird... 🤷‍♂️

miina reviewed (approved):
LGTM! (Pending the comments). Thanks for cleaning it up a bit, too.

Commit:
Fixed pointer placement again(?)

Commit:
Generalized color input and added tests for page background color

Commit:
Merge branch 'main' into fix/6804-rtl-popup

Commit:
Fixed import that caused a weird unrelated bug in a jest test

The bug was probably caused by some cyclic dependencies, but not really sure. This fixes it so, we are all good now!

Commit:
Fixed scrollbar visibility
